src/components/ComputeModal.js

```diff
@@ -195,16 +215,22 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
   })
   // State -- end
 
-  const isPersistentDisk = shouldUsePersistentDisk(sparkMode, currentRuntimeDetails, upgradeDiskSelected)
+  const isPersistentDisk = shouldUsePersistentDisk(runtimeType, currentRuntimeDetails, upgradeDiskSelected)
 
   const isCustomImage = selectedLeoImage === customMode
   const { version, updated, packages, requiresSpark, label: packageLabel } = _.find({ image: selectedLeoImage }, leoImages) || {}
 
-  const minRequiredMemory = sparkMode ? 7.5 : 3.75
+  const minRequiredMemory = isDataproc(runtimeType) ? 7.5 : 3.75
```

Comment:
Is there any information about why these two values were chosen for minRequiredMemory?

Reply:
Good question. I've added the comment below:

```
  // The memory sizes below are the minimum required to launch Terra-supported GCP runtimes, based on experimentation.
```

Does that clarify?

Reply:
Yes, thank you! Can't say how many times in my career that I've seen magic numbers in code and had no clue how someone arrived at them.

Reply:
💯